### ISSUE TITLE
Refactor Zip Creation as Part of CI/CD

### ISSUE THREAD

**Issue by osterman** [cloudposse/terraform-aws-ses-lambda-forwarder#2]
## what

Suggested improvements: (can be implemented in separate PR)
- [ ] Build/package zip as part of CI to create an artifact 
- [ ] Attach ZIP artifact to release
- [ ] Derive module version from git
- [ ] rename module to `terraform-aws-ses-lambda-forwarder`
- [ ] ensure S3 bucket uses encryption at rest (use https://github.com/cloudposse/terraform-aws-s3-log-storage for bucket creation)

## why
* Artifacts should be built from CI/CD and not by developers
* Artifacts should not be committed to source control

## tips

this appears to work to introspect the version of the current module:
```
variable "git_ref_type" {
  default = "tag"
}

data "external" "example" {
  count = "${var.git_ref_type == "tag" || var.git_ref_type == "branch" ? 1 : 0}"
  program = ["git", "-C", "${path.module}", "${var.git_ref_type}", "--points-at", "HEAD", "--format={\"ref\": \"%(refname:lstrip=2)\"}"]
}


output "ref" {
  value = "${join("", data.external.example.*.result.ref)}"
}
```

outputs:
```ref = test-0.1.1```

It fails if `git_ref` is not type "tag" or "branch"

**Comment by osterman** [comment]
- [ ] Add `ghr` package to `cloudposse/packages`
- [ ] Add `ghr` to `cloudposse/build-harness`

**Comment by osterman** [comment]
Evaluate: 
- <https://github.com/aktau/github-release>
- <https://github.com/tcnksm/ghr>

**Comment by osterman** [comment]
Related: https://github.com/cloudposse/build-harness/issues/82

**Comment by osterman** [comment]
Closed in #15
